test/export/test_torchbind.py

```diff
@@ -351,6 +414,188 @@ def forward(self, arg0_1, attr, arg1_1):
+    add_1 = torch.ops.aten.add.Tensor(add, arg1_1);  add = None
+    call_torchbind_2 = torch.ops.higher_order.call_torchbind(arg0_1, 'pop')
+    call_torchbind_3 = torch.ops.higher_order.call_torchbind(arg0_1, 'size')
+    sub = torch.ops.aten.sub.Tensor(call_torchbind_2, 0);  call_torchbind_2 = None
+    add_2 = torch.ops.aten.add.Tensor(sub, arg1_1);  sub = arg1_1 = None
+    return (add_2, add_1, arg0_1)
+    """,
+            )
+            # turn off tq type checking in eager execution
+            mod.check_tq_is_fake = False
+            self._assertEqualSkipScriptObject(gm(tq, x), mod(tq1, x))
+            self.assertEqual(tq.size(), 0)
+            self.assertEqual(tq1.size(), 0)
+
+
+@skipIfTorchDynamo("torchbind not supported with dynamo yet")
+class TestImplAbstractClass(TestCase):
+    def tearDown(self):
+        torch._library.fake_class_registry.global_fake_class_registry.clear()
+
+    def test_register_fake_class_no_torch_bind_class(self):
+        with self.assertRaisesRegex(RuntimeError, "Tried to instantiate class"):
+
+            @torch._library.register_fake_class("_TorchScriptTesting::NOT_A_VALID_NAME")
+            class Invalid:
+                pass
+
+    def test_register_fake_class_no_from_real(self):
+        with self.assertRaisesRegex(RuntimeError, "define a classmethod from_real"):
+
+            @torch._library.register_fake_class("_TorchScriptTesting::_Foo")
+            class InvalidFakeFoo:
+                def __init__(self):
+                    pass
+
+    def test_register_fake_class_from_real_not_classmethod(self):
+        with self.assertRaisesRegex(RuntimeError, "is not a classmethod"):
+
+            @torch._library.register_fake_class("_TorchScriptTesting::_Foo")
+            class FakeFoo:
+                def __init__(self, x, y):
+                    self.x = x
+                    self.y = y
+
+                def from_real(self, foo_obj):
+                    x, y = foo_obj.__getstate__()
+                    return FakeFoo(x, y)
+
+    def test_register_fake_class_valid(self):
+        class FakeFoo:
+            def __init__(self, x, y):
+                self.x = x
+                self.y = y
+
+            @classmethod
+            def from_real(cls, foo_obj):
+                x, y = foo_obj.__getstate__()
+                return cls(x, y)
+
+        torch._library.register_fake_class("_TorchScriptTesting::_Foo", FakeFoo)
+
+    def test_register_fake_class_duplicate_registration(self):
+        @torch._library.register_fake_class("_TorchScriptTesting::_Foo")
+        class FakeFoo:
+            def __init__(self, x, y):
+                self.x = x
+                self.y = y
+
+            @classmethod
+            def from_real(cls, foo_obj):
+                x, y = foo_obj.__getstate__()
+                return cls(x, y)
+
+        with self.assertRaisesRegex(RuntimeError, "already registered"):
+            torch._library.register_fake_class("_TorchScriptTesting::_Foo", FakeFoo)
```

Comment:
(no action required) If the user registers two, it's better to just accept the second one. I'm going to update `impl_abstract` to do that at some point. If it's not too difficult to make that happen in register_abstract_class now, we should.

Reply:
(registering two Fake classes should result in a warning but allow it through; this is useful for situations like colab)